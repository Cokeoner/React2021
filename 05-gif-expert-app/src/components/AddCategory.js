import React, {useState} from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ( {setCategories}) => {

    

    const [inputValue, setinputValue] = useState('')

    const handleInputChanche = (e) => {
        setinputValue( e.target.value );
    }


    const handleSubmit = (e) => {
        e.preventDefault();

        if(inputValue.trim().length > 2){
            setCategories(cats => [ ...cats, inputValue]);
            console.log('sdsdsd')
            setinputValue('');
        }

console.log('dsdsd')
        //console.log('submit hecho')
    }


    return (
        <form onSubmit={ handleSubmit }>

            <input 
                type = "Text"
                value = {inputValue}
                onChange={ (e) => handleInputChanche(e)}
            />

        </form>
    )
}

AddCategory.prototype = {
    setCategories: PropTypes.func.isRequired
}
