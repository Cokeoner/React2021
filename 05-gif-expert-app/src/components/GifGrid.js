import React from 'react'

export const GifGrid = ({category}) => {
   
    const getGif = () => {

        const url = 'api.giphy.com/v1/gifs/search?q=molotov&limit=10&api_key=GHgZQdqpRTEGzf8fzjkIXul7DwP2krj1';

    }
   
    return (
        <div>
            <li key = { category }> { category } </li>
        </div>
    )
}
