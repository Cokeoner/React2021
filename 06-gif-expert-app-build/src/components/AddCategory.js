import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({ setCategories }) => {



    const [inputValue, setinputValue] = useState('')

    const handleInputChanche = (e) => {
        setinputValue(e.target.value);

        console.log('HandleInputChange llamado');
    }


    const handleSubmit = (e) => {
        e.preventDefault();
        console.log('handleSubmit', inputValue)
        if (inputValue.trim().length > 2) {
            setCategories(cats => [inputValue, ...cats]);

            setinputValue('');
        }

        console.log('dsdsd')
        //console.log('submit hecho')
    }


    return (
        <form onSubmit={handleSubmit}>
            <p> {inputValue} </p>
            <input
                type="Text"
                value={inputValue}
                onChange={(e) => handleInputChanche(e)}
            />

        </form>
    )
}

AddCategory.prototype = {
    setCategories: PropTypes.func.isRequired
}
