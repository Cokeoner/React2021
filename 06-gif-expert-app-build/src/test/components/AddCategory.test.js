import { shallow } from "enzyme";
import { AddCategory } from "../../components/AddCategory";
import '@testing-library/jest-dom';

describe('Describbe la clase AddCategory', () => {

    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories} />);

    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories} />);
    })

    test('Debe de mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de cambbiar la caja de texto', () => {
        const input = wrapper.find('input');
        const value = 'Hola mundo';
        input.simulate('change', { target: { value } });

        expect(wrapper.find('p').text().trim()).toBe(value);
    });

    test('No debe de postear la informacion con submit', () => {

        wrapper.find('form').simulate('submit', { preventDefault() { } });
        expect(setCategories).not.toHaveBeenCalled();
    });


});