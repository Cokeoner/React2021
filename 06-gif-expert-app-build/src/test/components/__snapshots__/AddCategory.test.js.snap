// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Describbe la clase AddCategory Debe de mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p>
     
     
  </p>
  <input
    onChange={[Function]}
    type="Text"
    value=""
  />
</form>
`;
